hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -540,10 +566,17 @@ public void setTableVersion(HoodieTableVersion tableVersion) {
    * Read the payload class for HoodieRecords from the table properties.
    */
   public String getPayloadClass() {
-    // There could be tables written with payload class from com.uber.hoodie. Need to transparently
-    // change to org.apache.hudi
-    return getStringOrDefault(PAYLOAD_CLASS_NAME).replace("com.uber.hoodie",
-        "org.apache.hudi");
+    String payloadClassName;
+    if (contains(PAYLOAD_TYPE)) {
+      payloadClassName = RecordPayloadType.valueOf(getString(PAYLOAD_TYPE)).getClassName();
+    } else if (contains(PAYLOAD_CLASS_NAME)) {
+      payloadClassName = getString(PAYLOAD_CLASS_NAME);
+    } else {
+      payloadClassName = RecordPayloadType.valueOf(PAYLOAD_TYPE.defaultValue()).getClassName();
+    }
```

Comment:
It's better we move this method into the enum class itself, so that there is no similiar patched logic scattered everywhere.

Reply:
I don't think so, it's very hard to keep the logic int consistency because they are scatered all around, such code are likely to incur new bugs.

Reply:
hmm.. i think you are right. this has a chance of introducing bugs. 
i've refactored as suggested. please take a look at `KeyGeneratorType`, `RecordPayloadType`, `BootstrapIndexType`.